orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/aws/lambda/LambdaFunctionForceRefreshTask.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.providers.aws.lambda;
+
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.Task;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.clouddriver.CloudDriverCacheService;
+import com.netflix.spinnaker.orca.clouddriver.tasks.AbstractCloudProviderAwareTask;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import java.util.HashMap;
+import java.util.Map;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class LambdaFunctionForceRefreshTask extends AbstractCloudProviderAwareTask implements Task {
+
+  static final String REFRESH_TYPE = "Function";
+
+  public static final String TASK_NAME = "forceCacheRefresh";
+
+  @Autowired private CloudDriverCacheService cacheService;
+
+  @Override
+  public TaskResult execute(Stage stage) {
+    String cloudProvider = getCloudProvider(stage);
+
+    Map<String, Object> task = new HashMap<>(stage.getContext());
+    task.put("appName", stage.getExecution().getApplication());
+
+    cacheService.forceCacheUpdate(cloudProvider, REFRESH_TYPE, task);
+
+    return TaskResult.ofStatus(ExecutionStatus.SUCCEEDED);
```

Comment:
`forceCacheRefresh` is generally intended to pause pipeline progression until a given resource is actually visible via clouddriver. A fire-and-forget implementation isn't going to have the intended effect. See for example: https://github.com/spinnaker/orca/blob/master/orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/ServerGroupCacheForceRefreshTask.groovy

Reply:
Thanks for providing the details, we will delete this task. My intention is, CacheAgent will take few seconds to refresh the cache with new data, so I thought of using this task to refresh the cache instead of waiting for agent.

Reply:
If a lambda needs to be recent in clouddriver's cache in order to deploy, a proper cacheRefresh here is the right thing to do. But first, in clouddriver, `LambdaCachingAgent` needs to implement `OnDemandAgent`. Just calling `forceCacheUpdate` from Orca doesn't do anything, there'd be nothing to wait for currently.

Reply:
Yes, we already implemented the OnDemandAgent. Please check the code.

public class LambdaCachingAgent implements CachingAgent, AccountAware, OnDemandAgent

Reply:
`LambdaCachingAgent` in master does not implement `OnDemandAgent`: https://github.com/spinnaker/clouddriver/blob/master/clouddriver-lambda/src/main/java/com/netflix/spinnaker/clouddriver/lambda/provider/agent/LambdaCachingAgent.java - I do see that a clouddriver PR was opened to add this functionality last month but it fails the build and has a number of revision requests.

I suggest getting that PR sorted (https://github.com/spinnaker/clouddriver/pull/4048/files), then revise this PR to use it properly.